src/main/java/teammates/ui/webapi/action/SessionLinksRecoveryAction.java

```diff
@@ -0,0 +1,39 @@
+package teammates.ui.webapi.action;
+
+import teammates.common.util.Const;
+import teammates.common.util.EmailWrapper;
+import teammates.logic.api.EmailGenerator;
+import teammates.ui.webapi.output.SessionLinksRecoveryResponseData;
+
+
+/**
+ * Action specifically created for confirming email and sending session recovery links.
+ */
+public class SessionLinksRecoveryAction extends Action {
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.PUBLIC;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        // no specific access control needed.
+    }
+
+    @Override
+    public ActionResult execute() {
+        String recoveryEmailAddress = getNonNullRequestParamValue(Const.ParamsNames.SESSION_LINKS_RECOVERY_EMAIL);
+        boolean hasStudentsWithRecoveryEmail = !logic.getAllStudentForEmail(recoveryEmailAddress).isEmpty();
+
+        if (hasStudentsWithRecoveryEmail) {
+            EmailWrapper email = new EmailGenerator().generateSessionLinksRecoveryEmail(recoveryEmailAddress);
+            emailSender.sendEmail(email);
+        }
+
```

Comment:
Yes, we wont give out such information in the front-end because as mentioned [here](https://github.com/TEAMMATES/teammates/pull/9413#discussion_r260657191),

> We should not tell the user if the email is a valid TEAMMATES user because we don't want to leak that information to an unauthenticated user

In the front-end success message, we specify the email address to which the links were sent. So, the user can check with that if they entered a wrong email address.